towerlib/towerlib.py

```diff
@@ -1997,24 +2011,579 @@ def schedules(self):
+        """Get filtered list of hosts for a given inventory id.
+
+        Args:
+             inventory_id: the given inventory id.
+
+        Returns:
+            list: the list of filtered host as EntityManger object.
+
+        """
+        hosts = [item for item in self.hosts if item.inventory.id == inventory_id]
+        return hosts
+
+    def get_jobs_by_name(self, given_job_name):
+        """Get filtered list of jobs for a given name.
+
+        Args:
+            given_job_name: the given job name.
+
+        Returns:
+             list: the filtered list of jobs.
+
+        """
+        job_list = [item for item in self.jobs if item.name == given_job_name]
+        return job_list
+
+    def get_project_updates(self):
+        """Get all the project updates for the ansible tower.
+
+        Returns:
+            list: list of all the project updates.
+
+        """
+        project_updates_url = '{api}/project_updates/'.format(api=self.api)
+        response = self.session.get(project_updates_url)
+        if not response.ok:
+            self._logger.error("Error getting the project updates. response was: {})".format(response.text))
+            return None
+        else:
+            return response.json().get('results', [])
+
+    def get_project_updates_by_project_name(self, given_project_name):
+        """Get project update with the given project name.
+
+        Args:
+            given_project_name: the name of the project.
+
+        Returns:
+            dict: the project with all the information.
+
+        """
+        project_update_list = self.get_project_updates()
+        project_updates = [item for item in project_update_list if item['name'] == given_project_name]
+        return project_updates
+
+    def get_project_updates_by_project_id(self, given_project_id):
+        """Get project update with the given project id.
+
+        Args:
+            given_project_id: the id of the project.
+
+        Returns:
+            dict: the project with all the information.
+
+        """
+
+        project_update_list = self.get_project_updates()
+        project_updates = [item for item in project_update_list if item['project'] == given_project_id]
+        return project_updates
+
+    def get_project_update_by_id(self, project_update_id):
+        """Get project update with the given project_update id.
+
+        Args:
+            project_update_id: the id of the project_update.
+
+        Returns:
+            dict: the project_update with all the information.
+
+        """
+
+        project_updates_url = '{api}/project_updates/{id}'.format(api=self.api, id=project_update_id)
+        response = self.session.get(project_updates_url)
+        if not response.ok:
+            self._logger.error("Error getting the project updates. response was: {})".format(response.text))
+            return None
+        else:
+            return response.json()
+
+    def get_all_hosts_from_non_smart_inventories(self):
```

Comment:
I think that this can be achieved with a filter on the hosts objects.

Reply:
Could you pease write an example?

Currently my code looks like this: `        inventory_hosts = [item for item in self.hosts if item.inventory.kind != 'smart']
        return inventory_hosts`

Reply:
This can be achieved in client side. So, we skit it.